client.go

```diff
@@ -61,6 +61,8 @@ type Client struct {
 	cluster *Cluster
 	host    *URI
 	client  *http.Client
+	slice_clients map[string]*Client
```

Comment:
can we call this something like `clientCache` (also, underscores are not typically used in go)

and with that change, in `getDirectClient()` I would change `cacheClient` to just `client` and change `newClient :=` to `client =`.